app/config/ServiceConfiguration.scala

```diff
@@ -63,25 +64,57 @@ class PlayBasedServiceConfiguration @Inject()(configuration: Configuration, env:
   override def retryInterval = getRequired(configuration.getMilliseconds, "aws.sqs.retry.interval").milliseconds
 
   override def s3UrlExpirationPeriod(serviceName: String): FiniteDuration = {
-    val urlExpirationConfig =
-      s"${runMode.env}.upscan.consuming-services.${replaceInvalidJsonChars(serviceName)}.aws.s3.urlExpirationPeriod"
-    val expirationDuration: FiniteDuration =
-      getRequired(configuration.getMilliseconds, urlExpirationConfig).milliseconds
-
-    if (expirationDuration <= 7.days) {
-      expirationDuration
-    } else {
-      Logger.warn(
-        s"Expiration period for key: [$urlExpirationConfig] was: [$expirationDuration]. Using maximum value of 7 days instead.")
-      1.day
+    val configKey = configKeyForConsumingService(serviceName, S3UrlExpirationPeriod.ConfigDescriptor)
+    val maybeErrorOrExpiry = configuration.getMilliseconds(configKey).map(_.milliseconds).map(validateS3UrlExpirationPeriod(_,
```

Comment:
Types such as `Option[Either[_,_]]` always make me wonder if there's a different way of doing things...

I struggled to read the code in this line, and below. Can we switch to using Either or Option, and not both? And perhaps use a for-comprehension to keep the happy case obvious?

If it makes it easier, having less descriptive logging would be fine with me.